test/json.d

```diff
@@ -0,0 +1,308 @@
+    union
+    {
+        bool boolData;
+        BigInt bigIntData;
+        string stringData;
+        JsonObject objectData;
+        JsonValue[] arrayData;
+    }
+    JsonType type;
+    private this(JsonType type) { this.type = type; }
+    this(bool boolData) { this.boolData = boolData; this.type = JsonType.bool_; }
+    this(string stringData) { this.stringData = stringData; this.type = JsonType.string; }
+    this(BigInt bigIntData) { this.bigIntData = bigIntData; this.type = JsonType.number; }
+    this(JsonObject objectData) { this.objectData = objectData; this.type = JsonType.object; }
+    this(JsonValue[] arrayData) { this.arrayData = arrayData; this.type = JsonType.array; }
+    auto as(T)() inout
+    {
+        static if( is(T == bool) )
+        {
+            assert(type == JsonType.bool_, "expected json value to be a boolean but got " ~ type.to!string);
+            return boolData;
+        }
+        else static if( is(T == string) )
+        {
+            assert(type == JsonType.string, "expected json value to be a string but got " ~ type.to!string);
+            return stringData;
+        }
+        else static if( is(T == JsonValue[]) )
+        {
+            assert(type == JsonType.array, "expected json value to be an array but got " ~ type.to!string);
+            return arrayData;
+        }
+        else static if( is(T == JsonObject) )
+        {
+            assert(type == JsonType.object, "expected json value to be an object but got " ~ type.to!string);
+            return objectData;
+        }
+        else static assert(0);
+    }
+    auto array() inout { return as!(JsonValue[])(); }
+    auto object() inout { return as!(JsonObject)(); }
+}
+JsonValue parseJson(string text, string filenameForErrors)
+{
+    auto parser = JsonParser(text.ptr, text.ptr, filenameForErrors);
+    auto value = parser.parseValue();
+    parser.skipWhitespace();
+    if (*parser.next != '\0')
+        throw parser.parseError(parser.line(parser.next), "invalid json, multiple root values");
+    return value;
+}
+
+private bool isspace(char c)
+{
+    return c == ' ' || c == '\t' || c == '\r' || c == '\n';
+}
+private bool startsWith(const(char)* str, const(char)[] check)
+{
+    return startsWith(str, check.ptr, check.ptr + check.length);
+}
+private bool startsWith(const(char)* str, const(char)* check, const(char)* checkLimit)
+{
+    for (; check < checkLimit; str++, check++)
+    {
+        if(*str != *check)
+            return false;
+    }
+    return true;
+}
+class JsonParseException : Exception
+{
+    this(string msg, string file, size_t line)
+    { super(msg, file, line); }
+}
+struct JsonParser
+{
+    immutable(char)* start;
+    immutable(char)* next;
+    string filenameForErrors;
+    size_t line(immutable(char)* src)
+    {
+        size_t lineCount = 1;
+        for (auto countPtr = start; countPtr < next; countPtr++)
+        {
+            if (*countPtr == '\n')
+                lineCount++;
+        }
```

Comment:
How about: `text.filter!(a == '\n').walkLength`

Also there is `newline` in `std.ascii` and `std.uni`